util/afero.go

```diff
@@ -88,20 +88,30 @@ func stringify(fs afero.Fs, name string, indent []bool) (string, error) {
 // dst has a different set of contents in the same location, the behaviour is
 // currently undefined.
 // TODO: this should be made more rsync like and robust!
-func CopyFs(srcFs, dstFs afero.Fs, src, dst string, force bool) error {
-	if src == "" {
+func CopyFs(srcFs afero.Fs, dstFs afero.Fs, src string, dst string, force bool) error {
+	var srcFsLen int
+
+	if len(src) <= 1 {
+		// root directory
 		src = "/"
+		srcFsLen = 1
+	} else {
+		// todo: clean this up with function that gets parent dir?
+		src = strings.TrimSuffix(src, "/")
+		dirs := strings.Split(src, "/")
+		out := strings.Join(dirs[:len(dirs)-1], "/")
+		srcFsLen = len(out)
 	}
-	if dst == "" {
-		dst = "/"
+	if dst == "" || string(dst[len(dst)-1]) != "/" {
+		dst += "/"
 	}
+
 	walkFn := func(name string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
-		//perm := info.Perm()
-		perm := info.Mode() // TODO: is this correct?
-		p := path.Join(dst, filepath.Base(name))
+		perm := info.Mode() // get file permissions
```

Comment:
Is this correct?
I'm not suggesting it isn't, just wondering why you got rid of the TODO.

Reply:
i believe it is correct, which is why i removed the TODO.